File: sources/LiquidPlus/de/enzaxd/viaforge/platform/Platform.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  io.netty.util.concurrent.Future
 *  io.netty.util.concurrent.GenericFutureListener
 *  org.apache.logging.log4j.LogManager
 */
package de.enzaxd.viaforge.platform;

import com.viaversion.viaversion.api.command.ViaCommandSender;
import com.viaversion.viaversion.api.configuration.ConfigurationProvider;
import com.viaversion.viaversion.api.configuration.ViaVersionConfig;
import com.viaversion.viaversion.api.platform.PlatformTask;
import com.viaversion.viaversion.api.platform.ViaPlatform;
import com.viaversion.viaversion.libs.gson.JsonObject;
import com.viaversion.viaversion.libs.kyori.adventure.text.serializer.gson.GsonComponentSerializer;
import com.viaversion.viaversion.libs.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
import de.enzaxd.viaforge.ViaForge;
import de.enzaxd.viaforge.platform.ViaAPI;
import de.enzaxd.viaforge.platform.ViaConfig;
import de.enzaxd.viaforge.util.FutureTaskId;
import de.enzaxd.viaforge.util.JLoggerToLog4j;
import io.netty.util.concurrent.Future;
import io.netty.util.concurrent.GenericFutureListener;
import java.io.File;
import java.nio.file.Path;
import java.util.UUID;
import java.util.concurrent.CancellationException;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.TimeUnit;
import java.util.logging.Logger;
import org.apache.logging.log4j.LogManager;

public class Platform
implements ViaPlatform<UUID> {
    private final Logger logger = new JLoggerToLog4j(LogManager.getLogger((String)"ViaVersion"));
    private final ViaConfig config;
    private final File dataFolder;
    private final com.viaversion.viaversion.api.ViaAPI api;

    public Platform(File dataFolder) {
        Path configDir = dataFolder.toPath().resolve("ViaVersion");
        this.config = new ViaConfig(configDir.resolve("viaversion.yml").toFile());
        this.dataFolder = configDir.toFile();
        this.api = new ViaAPI();
    }

    public static String legacyToJson(String legacy) {
        return (String)GsonComponentSerializer.gson().serialize(LegacyComponentSerializer.legacySection().deserialize(legacy));
    }

    @Override
    public Logger getLogger() {
        return this.logger;
    }

    @Override
    public String getPlatformName() {
        return "ViaForge";
    }

    @Override
    public String getPlatformVersion() {
        return "47";
    }

    @Override
    public String getPluginVersion() {
        return "4.0.0";
    }

    @Override
    public FutureTaskId runAsync(Runnable runnable) {
        return new FutureTaskId((java.util.concurrent.Future<?>)((Object)CompletableFuture.runAsync(runnable, ViaForge.getInstance().getAsyncExecutor()).exceptionally(throwable -> {
            if (!(throwable instanceof CancellationException)) {
                throwable.printStackTrace();
            }
            return null;
        })));
    }

    @Override
    public FutureTaskId runSync(Runnable runnable) {
        return new FutureTaskId((java.util.concurrent.Future<?>)ViaForge.getInstance().getEventLoop().submit(runnable).addListener(this.errorLogger()));
    }

    @Override
    public PlatformTask runSync(Runnable runnable, long ticks) {
        return new FutureTaskId((java.util.concurrent.Future<?>)ViaForge.getInstance().getEventLoop().schedule(() -> this.runSync(runnable), ticks * 50L, TimeUnit.MILLISECONDS).addListener(this.errorLogger()));
    }

    @Override
    public PlatformTask runRepeatingSync(Runnable runnable, long ticks) {
        return new FutureTaskId((java.util.concurrent.Future<?>)ViaForge.getInstance().getEventLoop().scheduleAtFixedRate(() -> this.runSync(runnable), 0L, ticks * 50L, TimeUnit.MILLISECONDS).addListener(this.errorLogger()));
    }

    private <T extends Future<?>> GenericFutureListener<T> errorLogger() {
        return future -> {
            if (!future.isCancelled() && future.cause() != null) {
                future.cause().printStackTrace();
            }
        };
    }

    @Override
    public ViaCommandSender[] getOnlinePlayers() {
        return new ViaCommandSender[1337];
    }

    private ViaCommandSender[] getServerPlayers() {
        return new ViaCommandSender[1337];
    }

    @Override
    public void sendMessage(UUID uuid, String s) {
    }

    @Override
    public boolean kickPlayer(UUID uuid, String s) {
        return false;
    }

    @Override
    public boolean isPluginEnabled() {
        return true;
    }

    @Override
    public com.viaversion.viaversion.api.ViaAPI getApi() {
        return this.api;
    }

    @Override
    public ViaVersionConfig getConf() {
        return this.config;
    }

    @Override
    public ConfigurationProvider getConfigurationProvider() {
        return this.config;
    }

    @Override
    public File getDataFolder() {
        return this.dataFolder;
    }

    @Override
    public void onReload() {
    }

    @Override
    public JsonObject getDump() {
        JsonObject platformSpecific = new JsonObject();
        return platformSpecific;
    }

    @Override
    public boolean isOldClientsAllowed() {
        return true;
    }
}

File: sources/Kevin Reborn/kevin/via/Platform.kt
/*
 * This program is free software: you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation, either version 3 of the License, or
 *   (at your option) any later version.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package kevin.via

import com.viaversion.viaversion.api.command.ViaCommandSender
import com.viaversion.viaversion.api.platform.PlatformTask
import com.viaversion.viaversion.api.platform.ViaPlatform
import com.viaversion.viaversion.libs.gson.JsonObject
import com.viaversion.viaversion.libs.kyori.adventure.text.serializer.gson.GsonComponentSerializer
import com.viaversion.viaversion.libs.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer
import io.netty.util.concurrent.Future
import io.netty.util.concurrent.GenericFutureListener
import org.apache.logging.log4j.LogManager
import java.io.File
import java.util.*
import java.util.concurrent.CancellationException
import java.util.concurrent.CompletableFuture
import java.util.concurrent.TimeUnit

class Platform(dataFolder: File): ViaPlatform<UUID> {
    private val logger = JLoggerToLog4j(LogManager.getLogger("ViaVersion"))
    private var config: ViaConfig
    private val dataFolder: File
    init {
        val configDir = dataFolder.toPath().resolve("ViaVersion")
        config = ViaConfig(configDir.resolve("viaversion.yml").toFile())
        this.dataFolder = configDir.toFile()
    }
    private var api = ViaAPI()
    fun legacyToJson(legacy: String?) =
        GsonComponentSerializer.gson().serialize(LegacyComponentSerializer.legacySection().deserialize(legacy!!))
    override fun getLogger() = logger
    override fun getPlatformName() = "KevinClient"
    override fun getPlatformVersion() = ViaVersion.CLIENT_VERSION.toString()
    override fun getPluginVersion() = "4.0.0"
    override fun runAsync(runnable: Runnable?): FutureTaskId {
        return FutureTaskId(CompletableFuture
            .runAsync(runnable, ViaVersion.asyncExecutor)
            .exceptionally { throwable: Throwable ->
                if (throwable !is CancellationException) {
                    throwable.printStackTrace()
                }
                null
            }
        )
    }
    override fun runSync(runnable: Runnable): FutureTaskId {
        return FutureTaskId(
            ViaVersion.eventLoop!!.submit(runnable).addListener(errorLogger<Future<Any>>())
        )
    }
    override fun runSync(runnable: Runnable, ticks: Long): PlatformTask<*> {
        return FutureTaskId(
            ViaVersion.eventLoop!!.schedule(
                { runSync(runnable) }, ticks *
                        50, TimeUnit.MILLISECONDS
            ).addListener(errorLogger<Future<Any>>())
        )
    }
    override fun runRepeatingSync(runnable: Runnable, ticks: Long): PlatformTask<*> {
        return FutureTaskId(
            ViaVersion.eventLoop!!.scheduleAtFixedRate(
                { runSync(runnable) },
                0, ticks * 50, TimeUnit.MILLISECONDS
            ).addListener(errorLogger<Future<Any>>())
        )
    }
    private fun <T : Future<*>?> errorLogger(): GenericFutureListener<T> {
        return GenericFutureListener<T> { future ->
            if (!future!!.isCancelled && future.cause() != null) {
                future.cause().printStackTrace()
            }
        }
    }
    override fun getOnlinePlayers(): Array<ViaCommandSender?> = arrayOfNulls(1145)
    private fun getServerPlayers(): Array<ViaCommandSender?> = arrayOfNulls(1145)
    override fun sendMessage(uuid: UUID?, s: String?) {}
    override fun kickPlayer(uuid: UUID?, s: String?) = false
    override fun isPluginEnabled() = true
    override fun getApi() = api
    override fun getConf() = config
    override fun getConfigurationProvider() = config
    override fun getDataFolder() = dataFolder
    override fun onReload() {}
    override fun getDump() = JsonObject()
    override fun isOldClientsAllowed() = true
}